pbj-core/pbj-grpc-helidon/src/main/java/com/hedera/pbj/grpc/helidon/PbjProtocolSelector.java

```diff
@@ -0,0 +1,160 @@
+import io.helidon.http.Method;
+import io.helidon.http.Status;
+import io.helidon.http.WritableHeaders;
+import io.helidon.http.http2.FlowControl;
+import io.helidon.http.http2.Http2Flag;
+import io.helidon.http.http2.Http2FrameHeader;
+import io.helidon.http.http2.Http2Headers;
+import io.helidon.http.http2.Http2RstStream;
+import io.helidon.http.http2.Http2Settings;
+import io.helidon.http.http2.Http2StreamState;
+import io.helidon.http.http2.Http2StreamWriter;
+import io.helidon.http.http2.Http2WindowUpdate;
+import io.helidon.http.http2.StreamFlowControl;
+import io.helidon.metrics.api.Counter;
+import io.helidon.metrics.api.Metrics;
+import io.helidon.webserver.ConnectionContext;
+import io.helidon.webserver.Router;
+import io.helidon.webserver.http2.spi.Http2SubProtocolSelector;
+import io.helidon.webserver.http2.spi.SubProtocolResult;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Sub-protocol selector for HTTP/2. This is the main entry point into the PBJ implementation of gRPC. The web
+ * server will use this class to determine if a request is a gRPC request and if so, how to handle it.
+ */
+class PbjProtocolSelector implements Http2SubProtocolSelector {
+    /** This routing has absolutely no routes. */
+    private static final PbjRouting EMPTY = PbjRouting.builder().build();
+
+    private final PbjConfig config;
+    private final DeadlineDetector deadlineDetector;
+    private final ScheduledExecutorService deadlineExecutorService = Executors.newSingleThreadScheduledExecutor();
+    private final Counter requestCounter;
+    private final Counter failedRequestCounter;
+
+    /**
+     * Create a new PBJ based grpc protocol selector (default). Access restricted to be package-private so as
+     * to limit instantiation to the {@link PbjProtocolProvider}.
+     */
+    PbjProtocolSelector(@NonNull final PbjConfig config) {
+        this.config = requireNonNull(config);
+        this.deadlineDetector = (deadline, onDeadlineExceeded) ->
+                deadlineExecutorService.schedule(onDeadlineExceeded, deadline, TimeUnit.NANOSECONDS);
+
+        final var metricRegistry = Metrics.globalRegistry();
+        this.requestCounter = metricRegistry.getOrCreate(Counter.builder("pbj.grpc.requests")
+                .scope("vendor")
+                .description("The number of gRPC requests"));
+        this.failedRequestCounter = metricRegistry.getOrCreate(Counter.builder("pbj.grpc.request.failures")
+                .scope("vendor")
+                .description("The number of failed gRPC requests"));
+    }
+
+    /**
+     * Called by Helidon to create the sub-protocol for PBJ gRPC requests. The {@link SubProtocolResult} returned
+     * will be responsible for handling the request.
+     */
+    @Override
+    public SubProtocolResult subProtocol(ConnectionContext ctx,
+                                         HttpPrologue prologue,
```

Comment:
Minor: I suggest to move the first argument to a new line with a regular continuation indentation of 8 spaces and avoid a huge block of vertical whitespace here for readability.